Sources/Sentry/SentrySession.m

```diff
@@ -22,110 +22,120 @@ - (instancetype)init {
-    }
+        if (nil != statusString) {
+            [serializedData setValue:statusString forKey:@"status"];
+        }
 
-    if (nil != _init) {
-        [serializedData setValue:_init forKey:@"init"];
-    }
+        if (nil != _timestamp) {
+            [serializedData setValue:[_timestamp sentry_toIso8601String] forKey:@"timestamp"];
+        }
 
-    if (nil != _duration) {
-        NSNumber *duration = [NSNumber numberWithLongLong:[[NSDate date] timeIntervalSince1970] * 1000];
-        [serializedData setValue:duration forKey:@"duration"];
-    }
+        if (nil != _init) {
+            [serializedData setValue:_init forKey:@"init"];
+        }
 
-    if (nil != _sequence) {
-        [serializedData setValue:[@(*_sequence) stringValue] forKey:@"seq"];
-    }
+        if (nil != _duration) {
+            NSNumber *duration = [NSNumber numberWithLongLong:[[NSDate date] timeIntervalSince1970] * 1000];
+            [serializedData setValue:duration forKey:@"duration"];
+        }
 
-    // TODO: Add the following under `attrs`. Except 'did'
-    if (nil != _releaseName) {
-        [serializedData setValue:_releaseName forKey:@"release"];
-    }
+        if (nil != _sequence) {
+            [serializedData setValue:[@(*_sequence) stringValue] forKey:@"seq"];
+        }
 
-    if (nil != _environment) {
-        [serializedData setValue:_environment forKey:@"environment"];
-    }
+        // TODO: Add the following under `attrs`. Except 'did'
+        if (nil != _releaseName) {
+            [serializedData setValue:_releaseName forKey:@"release"];
+        }
 
-    SentryUser *currentUser = _user;
-    NSString *did = _distinctId;
-    if (nil != currentUser) {
-        NSString *ipAddress = currentUser.ipAddress;
-        if (nil != ipAddress) {
-            [serializedData setValue:ipAddress forKey:@"ip_address"];
+        if (nil != _environment) {
+            [serializedData setValue:_environment forKey:@"environment"];
         }
-        if (nil == did) {
-            if (nil != currentUser.userId) {
-                did = currentUser.userId;
-            } else if (nil != currentUser.email) {
-                did == currentUser.email;
-            } else if (nil != currentUser.username) {
-                did == currentUser.username;
+
+        SentryUser *currentUser = _user;
+        NSString *did = _distinctId;
+        if (nil != currentUser) {
+            NSString *ipAddress = currentUser.ipAddress;
+            if (nil != ipAddress) {
+                [serializedData setValue:ipAddress forKey:@"ip_address"];
+            }
+            if (nil == did) {
+                if (nil != currentUser.userId) {
+                    did = currentUser.userId;
+                } else if (nil != currentUser.email) {
+                    did == currentUser.email;
+                } else if (nil != currentUser.username) {
+                    did == currentUser.username;
+                }
             }
```

Comment:
I don't do this on Android, if `distinctId` is not set, I don't send at all, the reason is to avoid complaints about sensitive data as there's no other way around of PII it.

Reply:
This I saw in python: https://github.com/getsentry/sentry-python/blob/master/sentry_sdk/sessions.py#L184-L189

Reply:
yep, it was like this on Android as well, we've decided to change it, this was made after python impl. iirc, so we always use a generated GUID and cache it on disk.
downside is, if a user uninstall/install the app, a new GUID will be generated.

Reply:
got it. This approach on python anyway looks more like a server-side thing,
I'll change to use the install id. 👍

Reply:
In the python SDK this should take the `send_default_pii` thing into account. Will fix that. Same here i think.

Reply:
For `did` we just use the `installation id` to uniquely identify the device. It's like that on Android and makes sense here too

Reply:
On second thought this does not need to be changed on the Python side as the user would not be set in the first place if the PII flag is not true.

Reply:
We'll do device id as in Android